cdap-docs/developer-manual/source/pipelines/developing-plugins/plugin-basics.rst

```diff
@@ -126,3 +126,14 @@ In this example, we have a plugin of type ``batchsource``, named ``MyBatchSource
 This plugin takes two configuration properties. The first is named ``input-path`` and is required.
 The second is named ``cleanOutput`` and is optional. Note that optional configuration fields should
 have their default values set in the no-argument constructor.
+
+Plugin Data types
+=================
+
+Plugins support various data types such as boolean, bytes, double, float, int, long, string, array, enum, map, union
+and record. Temporal data types such as date, time and timestamp are also supported. Date represents number of days
```

Comment:
The audience for this page are plugin developers so we can be more technical and talk about the actual logical types.

Plugins support several physical data types: null, boolean, int, long, float, double, string, array, map, enum, union, and record. They also support several logical data types: date, time-millis, time-micros, timestamp-millis, and timestamp-micros. Logical types represent some abstract concept but are internally represented with a corresponding physical type. The date logical type is an integer that holds the number of days since the Unix epoch. The time type is a long that holds the number of milliseconds or microseconds after midnight. The timestamp type is a long that holds the number of milliseconds or microseconds since the Unix epoch.

To create a schema field of a logical type:
[code sample]

To set a record field for a logical type:
[code sample]

To read a logical type from a record:
[code sample]